clients/vault/src/metrics.rs

```diff
@@ -1,27 +1,1341 @@
+			.expect("Failed to create prometheus metric");
+	pub static ref COLLATERALIZATION: GaugeVec =
+		GaugeVec::new(Opts::new("collateralization", "Collateralization"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
+	pub static ref REQUIRED_COLLATERAL: GaugeVec =
+		GaugeVec::new(Opts::new("required_collateral", "Required Collateral"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
+	pub static ref MEAN_IDLE_DURATION: IntGaugeVec =
+		IntGaugeVec::new(Opts::new("mean_idle_duration_ms", "Total Idle Duration"), &[TASK_NAME])
+			.expect("Failed to create prometheus metric");
+	pub static ref MEAN_POLL_DURATION: IntGaugeVec =
+		IntGaugeVec::new(Opts::new("mean_poll_duration_ms", "Total Poll Duration"), &[TASK_NAME])
+			.expect("Failed to create prometheus metric");
+	pub static ref MEAN_SCHEDULED_DURATION: IntGaugeVec = IntGaugeVec::new(
+		Opts::new("mean_scheduled_duration_ms", "Total Scheduled Duration"),
+		&[TASK_NAME]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref UTXO_COUNT: IntGaugeVec = IntGaugeVec::new(
+		Opts::new("utxo_count", "Number of Unspent Bitcoin Outputs"),
+		&[CURRENCY_LABEL]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref BTC_BALANCE: GaugeVec = GaugeVec::new(
+		Opts::new("btc_balance", "Bitcoin Balance"),
+		&[CURRENCY_LABEL, BTC_BALANCE_TYPE_LABEL]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref ISSUES: GaugeVec = GaugeVec::new(
+		Opts::new("issue_count", "Number of issues"),
+		&[CURRENCY_LABEL, REQUEST_STATUS_LABEL]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref REDEEMS: GaugeVec = GaugeVec::new(
+		Opts::new("redeem_count", "Number of redeems"),
+		&[CURRENCY_LABEL, REQUEST_STATUS_LABEL]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref NATIVE_CURRENCY_BALANCE: Gauge =
+		Gauge::new("native_currency_balance", "Native Currency Balance")
+			.expect("Failed to create prometheus metric");
+	pub static ref FEE_BUDGET_SURPLUS: GaugeVec =
+		GaugeVec::new(Opts::new("fee_budget_surplus", "Fee Budget Surplus"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
 	pub static ref RESTART_COUNT: IntCounter =
 		IntCounter::new("restart_count", "Number of service restarts")
 			.expect("Failed to create prometheus metric");
 }
 
+#[derive(Clone, Debug)]
+struct AverageTracker {
+	total: u64,
+	count: u64,
+}
+
+#[derive(Clone, Debug)]
+struct StatefulGauge<T: Clone> {
+	gauge: Gauge,
+	data: Arc<RwLock<T>>,
+}
+
+#[derive(Clone, Debug)]
+struct XLMBalance {
+	upperbound: Gauge,
+	lowerbound: Gauge,
+	actual: Gauge,
+}
+
+#[derive(Clone, Debug)]
+struct RequestCounter {
+	open_count: Gauge,
+	completed_count: Gauge,
+	expired_count: Gauge,
+}
+
+#[derive(Clone, Debug)]
+pub struct PerCurrencyMetrics {
+	locked_collateral: Gauge,
+	collateralization: Gauge,
+	required_collateral: Gauge,
+	remaining_time_to_redeem_hours: Gauge,
+	btc_balance: XLMBalance,
```

Comment:
this should be renamed to `lumen_balance` or `xlm_balance`

Reply:
Done